src/main/java/request/HttpRequestBuilder.java

```diff
@@ -9,12 +10,33 @@ public HttpRequestBuilder(IValidator requestVerifier) {
     this.requestVerifier = requestVerifier;
   }
 
-  public HttpRequest createRequest(String rawRequest) throws InvalidHttpRequestException {
-    String rawRequestArray[] = rawRequest.split("\n", 2);
+  public HttpRequest createRequest(String rawRequest) throws IOException, InvalidHttpRequestException {
+    HttpRequest rtnRequest;
+    String requestLine = parseRequestLine(rawRequest);
+    HashMap<String, String> headers = parseHeaders(rawRequest);
+    rtnRequest = new HttpRequest(requestLine, headers);
+    return rtnRequest;
+  }
+
+  public String parseRequestLine(String rawRequest) throws InvalidHttpRequestException {
     String requestLine = rawRequest.split("\r?\n")[0];
     if (!requestVerifier.isValid(requestLine)) {
       throw new InvalidHttpRequestException();
     }
-    return new HttpRequest(requestLine);
+    return requestLine;
+  }
+
+  public HashMap<String, String> parseHeaders(String rawRequest) throws InvalidHttpRequestException  {
+    String rawHeaders = rawRequest.split("\n", 2)[1];
+    HashMap<String, String> rtnMap = new HashMap<String, String>();
+    String headerLines[] = rawHeaders.split("\n");
+    for (String line : headerLines ) {
+      if (line.contains(":") && line.contains(" ")) {
+        String key = line.split(":", 2)[0];
```

Comment:
Can we do a `split(": ")` in java (with the whitespace already included in the split-by-string? Would that avoid this weird double-split?

Also, could probably do the split right away and then have a `if (splitThing.length == 2)` instead of the weird `contains` logic.

(I'm not actually sure that you can split by a multi-character string.)